File: C#/Merge Two Sorted Lists/Merge2SortedLists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode MergeTwoLists (ListNode l1, ListNode l2) {
        ListNode head = new ListNode (0);
        ListNode curr = head;
        while (l1 != null && l2 != null) {
            if (l1.val <= l2.val) {
                curr.next = l1;
                l1 = l1.next;
            } else {
                curr.next = l2;
                l2 = l2.next;
            }
            curr=curr.next;
        }
        if (l1 != null) {
            curr.next = l1;
            l1 = l1.next;
        } else {
            if (l2 != null) {
                curr.next = l2;
                l2 = l2.next;
            } else {
                curr.next = null;
            }
        }
        return head.next;
    }
}

File: Python/Merge Two Sorted Lists/Merge2SortedLists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        l1_val = []
        l2_val = []
        while l1:
            l1_val.append(l1.val)
            l1 = l1.next
        while l2:
            l2_val.append(l2.val)
            l2 = l2.next
        val_list = sorted(l1_val + l2_val)
        
        result = head = ListNode(0)
        for i in val_list:
            head.next = ListNode(i)
            head = head.next
        head.next = None

        return result.next